core/Decode.cpp

```diff
@@ -131,37 +127,28 @@ namespace olympia
         }
     }
 
-    void Decode::updateVcsrs_(const InstPtr & inst)
+    void Decode::updateVectorConfig_(const InstPtr & inst)
     {
-        VCSRs_.setVCSRs(inst->getVL(), inst->getSEW(), inst->getLMUL(), inst->getVTA());
+        vector_config_ = inst->getVectorConfig();
 
+        // If rs1 is x0 and rd is x0 then the vl is unchanged (assuming it is legal)
         const uint64_t uid = inst->getOpCodeInfo()->getInstructionUniqueID();
-        if ((uid == mavis_vsetvli_uid_) && inst->hasZeroRegSource())
+        if ((uid == MAVIS_UID_VSETVLI) && inst->hasZeroRegSource())
         {
-            // If rs1 is x0 and rd is x0 then the vl is unchanged (assuming it is legal)
-            VCSRs_.vl = inst->hasZeroRegDest() ? std::min(VCSRs_.vl, VCSRs_.vlmax)
-                                               : VCSRs_.vlmax;
+            const uint32_t new_vl = \
+                inst->hasZeroRegDest() ? std::min(vector_config_->getVL(), vector_config_->getVLMAX())
+                                       : vector_config_->getVLMAX();
+            vector_config_->setVL(new_vl);
         }
 
-        ILOG("Processing vset{i}vl{i} instruction: " << inst);
-        ILOG("  LMUL: " << VCSRs_.lmul);
-        ILOG("   SEW: " << VCSRs_.sew);
-        ILOG("   VTA: " << VCSRs_.vta);
-        ILOG(" VLMAX: " << VCSRs_.vlmax);
-        ILOG("    VL: " << VCSRs_.vl);
-
-        // Check validity of vector config
-        sparta_assert(VCSRs_.lmul <= 8,
-            "LMUL (" << VCSRs_.lmul << ") cannot be greater than " << 8);
-        sparta_assert(VCSRs_.vl <= VCSRs_.vlmax,
-            "VL (" << VCSRs_.vl << ") cannot be greater than VLMAX ("<< VCSRs_.vlmax << ")");
+        ILOG("Processing vset{i}vl{i} instruction: " << inst << " " << vector_config_);
     }
 
     // process vset settings being forward from execution pipe
     // for set instructions that depend on register
     void Decode::process_vset_(const InstPtr & inst)
```

Comment:
Do you mind changing the method name to match the style guide?

Reply:
Done.